### PR TITLE
allow unstake if emergency unlock is triggered

### PR DESCRIPTION
**Pull request by gbonumore**
##### Description
This allows unstake if emergency unlock is triggered

### CODE REVIEW

File: src/components/staking/Positions.svelte [hunk 1]
```diff
@@ -111,9 +114,10 @@
 
                         boostToMax(lock.lockId, $eth)
                           .then((updated_data) => {
-                            $timestampBoosted[updated_data.accountLocks[0].lockId] = boostingTimestamp;
-
-                            if(scrollToTop) {
+                            $timestampBoosted[
+                              updated_data.accountLocks[0].lockId
```
Comment (jordaniza) [inline src/components/staking/Positions.svelte:118]: will this break if accountLocks is len zero?

File: src/helpers/staking.js [hunk 2]
```diff
@@ -237,6 +237,20 @@ export async function getLastLockForAddress(eth) {
   }
 }
 
+export async function getEmergencyUnlock(eth) {
+  try {
+    if (!sharesTimeLock) {
+      initContracts(eth);
```
Comment (jordaniza) [inline src/helpers/staking.js:243]: does init contracts need to be async?

Comment (gbonumore) [reply src/helpers/staking.js:243]: it doesn't, creating a new contract instance is not an async operation

File: src/components/staking/Positions.svelte (hunk 1 above)
Comment (gbonumore) [reply src/components/staking/Positions.svelte:118]: even if this is some lint edit, as far as I can read, this code comes after a boostToMax so it's unlikely it has no locks, it fetches data again after that

File: src/helpers/staking.js (hunk 2 above)
Comment (jordaniza) [reply src/helpers/staking.js:243]: It surprises me that this code works - if we don't have the sharesTimelock, how can we run the rest of the code block as we are missing data from it? And if it takes some time to fetch the shares timelock - wouldn't it break everything?

OTOH - if you're happy it works, then no need to spend ages on it.

**Pull request merged**